### PR TITLE
cats-effect Engine implementation for tests

### PR DESCRIPTION
**Pull request by dfakhritdinov**

### CODE REVIEW

File: eventsourcing/src/main/scala/com/evolutiongaming/akkaeffect/eventsourcing/Engine.scala [hunk 1]
```diff
@@ -266,6 +265,80 @@ object Engine {
     } yield engine
   }
 
+  /**
+   * Cats-effect based implementation, expected to be used '''only in tests'''
+   */
+  def of[F[_]: Async, S, E](
```
Comment (t3hnar) [inline eventsourcing/src/main/scala/com/evolutiongaming/akkaeffect/eventsourcing/Engine.scala:271]: does it work anyhow differently than the original one?

Comment (dfakhritdinov) [reply eventsourcing/src/main/scala/com/evolutiongaming/akkaeffect/eventsourcing/Engine.scala:271]: I would say no, but I have not tested it properly. Main difference is that no akka required

File: eventsourcing/src/main/scala/com/evolutiongaming/akkaeffect/eventsourcing/Engine.scala [hunk 2]
```diff
@@ -333,5 +405,26 @@ object Engine {
...
+
+    trait Mock[F[_], A] {
+      def events: F[List[A]]
+    }
+
+
+    def mock[F[_]: Sync, A](initial: SeqNr): F[Append[F, A] with Mock[F, A]] =
+      for {
+        ref <- Ref.of[F, List[A]](List.empty)
+      } yield new Append[F, A] with Mock[F, A] {
```
[the remark this answers is not in the document]
Comment (dfakhritdinov) [reply eventsourcing/src/main/scala/com/evolutiongaming/akkaeffect/eventsourcing/Engine.scala:418]: What's the benefits of having `Append.apply(ref: Ref[List[E])`? I tried to hide implementation details for usage simplicity

File: eventsourcing/src/main/scala/com/evolutiongaming/akkaeffect/eventsourcing/Engine.scala (hunk 1 above)
Comment (t3hnar) [reply eventsourcing/src/main/scala/com/evolutiongaming/akkaeffect/eventsourcing/Engine.scala:271]: will it be hard - using existing unit test for both cases: akka based and this impls?

File: eventsourcing/src/main/scala/com/evolutiongaming/akkaeffect/eventsourcing/Engine.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (t3hnar) [reply eventsourcing/src/main/scala/com/evolutiongaming/akkaeffect/eventsourcing/Engine.scala:418]: maybe

[the remark this answers is not in the document]
Comment (t3hnar) [reply eventsourcing/src/main/scala/com/evolutiongaming/akkaeffect/eventsourcing/Engine.scala:418]: however I really do not like that you mix `Append` & `Mock` into the same instance

[the remark this answers is not in the document]
Comment (dfakhritdinov) [reply eventsourcing/src/main/scala/com/evolutiongaming/akkaeffect/eventsourcing/Engine.scala]: done

File: eventsourcing/src/main/scala/com/evolutiongaming/akkaeffect/eventsourcing/Engine.scala (hunk 1 above)
Comment (dfakhritdinov) [reply eventsourcing/src/main/scala/com/evolutiongaming/akkaeffect/eventsourcing/Engine.scala:288]: I've improved the implementation to be inline with already existing Akka based impl

File: eventsourcing/src/test/scala/com/evolutiongaming/akkaeffect/eventsourcing/EngineAkkaTest.scala [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import cats.effect.implicits.effectResourceOps
+import cats.effect.unsafe.implicits.global
+import com.evolutiongaming.akkaeffect.ActorSuite
+import com.evolutiongaming.akkaeffect.IOSuite._
+import com.evolutiongaming.catshelper.{FromFuture, ToFuture}
+import org.scalatest.funsuite.AsyncFunSuite
+
+class EngineAkkaTest extends AsyncFunSuite with EngineTestCases with ActorSuite {
+
+  test("order of stages") {
```
Comment (t3hnar) [inline eventsourcing/src/test/scala/com/evolutiongaming/akkaeffect/eventsourcing/EngineAkkaTest.scala:15]: could we make this copied list of tests also part of EngineTestCases?

**Review by t3hnar — CHANGES_REQUESTED** [review]

**Pull request merged**